### ISSUE TITLE
esc overload activates after using mouse scrollwheel

### ISSUE THREAD

**Issue by i-c-u-p** [rvaiya/keyd#191]
When holding down a key that's overloaded to a layer on hold and esc on release (e.g. `capslock = overload(CapsLayer, esc)`) when the mouse wheel is scrolled and then the overloaded key is released the esc overload will activate anyway. For my config (below), here's how I noticed it:

- Open a browser, go to a site with a textbox and click it so keystrokes go to it when typing
- Hold down capslock and use the scroll wheel to produce Ctrl+mwheelup/down inputs
- Let go of capslock, which activates escape, causing the keyboard to lose focus of the text box

It seems to only happen with the scrollwheel as the esc overload doesn't activate after pressing left/right click or my mouse's side buttons for navigation.

keyd version: commit 6c4c6ac (test branch)

<details>
<summary>
/etc/keyd/default.conf:
</summary>

```
[ids]
*

# Colemak-DH (Curl+Angle mods)
[main]
esc = overload(LayoutLayer, capslock)
capslock = overload(CapsLayer, esc)
rightmeta = compose
rightcontrol = rightcontrol
rightshift = rightshift
e = f
r = p
t = b
y = j
u = l
i = u
o = y
p = ;
s = r
d = s
f = t
h = m
j = n
k = e
l = i
; = o
z = x
x = c
c = d
b = z
n = k
m = h

[QWERTY]
e = e
r = r
t = t
y = y
u = u
i = i
o = o
p = p
s = s
d = d
f = f
h = h
j = j
k = k
l = l
; = ;
z = z
x = x
c = c
b = b
n = n
m = m

[NoCapsLayer]
capslock = esc

[NumPrac]
a = tab
s = S-enter
d = C-backspace
f = backspace

[CapsLayer:C]
j = left
k = down
l = up
; = right
1 = f1
2 = f2
3 = f3
4 = f4
5 = f5
6 = f6
7 = f7
8 = f8
9 = f9
0 = f10
- = f11
= = f12
m = home
/ = end
i = pagedown
o = pageup
f = backspace
r = delete
space = layer(control)
d = layer(shift)
s = layer(alt)
g = enter

[CapsLayer+control]
s = layer(alt)
d = layer(shift)

[CapsLayer+shift]
space = layer(control)
s = layer(alt)

[CapsLayer+alt]
d = layer(shift)
space = layer(control)

[CapsLayer+control+shift]
s = layer(alt)

[CapsLayer+control+alt]
d = layer(shift)

[CapsLayer+shift+alt]
space = layer(control)

[CapsLayer+control+shift+alt]

[LayoutLayer]
q = toggle(QWERTY)
w = toggle(NoCapsLayer)
e = toggle(NumPrac)
```

</details>

**Comment by rvaiya** [comment]
I believe this should work in the latest commit.

**Comment by i-c-u-p** [comment]
> I believe this should work in the latest commit.

Looks like it's fixed, thanks!

**Comment by i-c-u-p** [comment]
This bug is back as of keyd 0c91ed8.

default.conf:
```
[ids]
*

[main]
capslock = overload(CapsLayer, esc)

[CapsLayer:C]
```

**Comment by rvaiya** [comment]
Thanks. I made some changes to the mouse handling logic. The latest commit should add a fix for this, though you will now need to explicitly list the mouse id in your config. The old approach was a bit of hack with several limitations.

**Comment by rvaiya** [comment]
> theres 2 scroll wheel inputs instead of 1 for each scroll
> Edit: the side buttons for navigation also don't work

Can you post the output of `keyd -m` after scrolling one notch and pressing one of these buttons? Do this while keyd is both running and not running.

> and my mouse sensitivity settings set with xinput set-prop are ignored.

If you add your mouse to the config all events will be routed through keyd's virtual pointer. I intend to add mouse-specific options at some point to allow things like sensitivity to be set within keyd, but for the moment you can just use xinput on keyd's virtual pointer.

**Comment by i-c-u-p** [comment]
> > theres 2 scroll wheel inputs instead of 1 for each scroll
> > Edit: the side buttons for navigation also don't work
> 
> Can you post the output of `keyd -m` after scrolling one notch and pressing one of these buttons? Do this while keyd is both running and not running.
> 
> > and my mouse sensitivity settings set with xinput set-prop are ignored.
> 
> If you add your mouse to the config all events will be routed through keyd's virtual pointer. I intend to add mouse-specific options at some point to allow things like sensitivity to be set within keyd, but for the moment you can just use xinput on keyd's virtual pointer.

i tested with 2 mice and the output seems to be the same without keyd running, with it running, and after adding the id's to default.conf (other than the "keyd virtual pointer" showing up instead of the mouse names). the side button thats closer to left click and goes forward shows up as mouse2 and the one thats further and goes backward shows up as mouse1.

Setting the sensitivity of keyd's virtual pointer instead of my actual mice in xinput works, although i normally use different sensitivity settings for both mice. i don't switch between them often so i'll just alias a toggle.

**Comment by rvaiya** [comment]
The scroll and button issues should be fixed in the latest commit. 

> Setting the sensitivity of keyd's virtual pointer instead of my actual mice in xinput works, although i normally use different sensitivity settings for both mice. i don't switch between them often so i'll just alias a toggle.

Once sensitivity can be set in keyd this should be less of an issue. I will some time to think about what I want mouse config syntax to look like before implementing this.

**Comment by i-c-u-p** [comment]
> The scroll and button issues should be fixed in the latest commit.
> 
> > Setting the sensitivity of keyd's virtual pointer instead of my actual mice in xinput works, although i normally use different sensitivity settings for both mice. i don't switch between them often so i'll just alias a toggle.
> 
> Once sensitivity can be set in keyd this should be less of an issue. I will some time to think about what I want mouse config syntax to look like before implementing this.

Both issues are fixed after updating to 6064708
